Java/jni/native_glue.hh

```diff
@@ -34,7 +34,11 @@ public:
     { s._slice = slice::null; }
 
     operator slice()    {return _slice;}
+#ifdef _MSC_VER
+    operator C4Slice()  {return {_slice.buf, _slice.size};}
```

Comment:
You don't need to conditionalize this. Just change the existing implementation to use the standard syntax.